### PR TITLE
Store `index.version.created` at partitioned table template

### PR DESCRIPTION
**Pull request by seut**
## Summary of the changes / Why this improves CrateDB

In order to expose the correct version a partitioned table was created on, we must store it inside the partitioned table
template settings. Already created partitioned tables will use the existing incorrect logic of showing the current used version at`information_schema.tables.version.created`.

This only relates to a partitioned table as a whole and not to dedicated partitions.

Relates to #12853 but won't fix it for existing tables as it isn't possible.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/appendices/release-notes/unreleased.rst [hunk 1]
```diff
@@ -130,3 +130,9 @@ Fixes
 
 - Fixed an issue that caused ``array_upper`` and ``array_lower`` scalar
   functions return wrong results on multidimensional arrays.
+
+- Fixed exposure of the correct version a partitioned table was created on at
+  the ``information_schema.tables.version.created`` column for newly created
+  partitioned tables. This won't fix it for existing tables as the version
+  information the table was created with was not stored and such is not
+  known.
```
Comment (matriv) [inline docs/appendices/release-notes/unreleased.rst:138]: To be more clear to the users, maybe add: `The issue doesn't affect the version exposed for individual partitions.` ?

File: server/src/test/java/io/crate/metadata/doc/DocTableInfoTest.java [hunk 2]
```diff
@@ -98,27 +94,27 @@ public void testGetColumnInfo() throws Exception {
...
         DynamicReference reference = info.getDynamic(col, false, true);
-        assertNull(reference);
+        assertThat(reference).isNull();
 
         // forWrite: true, errorOnUnknownObjectKey: true, parentPolicy: dynamic
         reference = info.getDynamic(col, true, true);
-        assertNotNull(reference);
-        assertSame(reference.valueType(), DataTypes.UNDEFINED);
+        assertThat(reference).isNotNull();
+        assertThat(reference.valueType()).isSameAs(DataTypes.UNDEFINED);
```
Comment (matriv) [inline server/src/test/java/io/crate/metadata/doc/DocTableInfoTest.java:106]: Could `isSameAs` be replaced with `isEqualTo()`?

File: server/src/test/java/io/crate/metadata/doc/DocTableInfoTest.java [hunk 3]
```diff
@@ -224,9 +220,22 @@ public void test_can_retrieve_all_parents_of_nested_object_column() throws Excep
 
         TableInfo table = e.resolveTableInfo("tbl");
         Iterable<Reference> parents = table.getParents(new ColumnIdent("o1", List.of("o2", "x")));
-        assertThat(parents, contains(
+        assertThat(parents).contains(
```
Comment (matriv) [inline server/src/test/java/io/crate/metadata/doc/DocTableInfoTest.java:223]: `containsExactly` ?

File: server/src/test/java/io/crate/metadata/doc/DocTableInfoTest.java [hunk 4]
```diff
@@ -224,9 +220,22 @@ public void test_can_retrieve_all_parents_of_nested_object_column() throws Excep
...
             table.getReference(new ColumnIdent("o1", "o2")),
             table.getReference(new ColumnIdent("o1"))
-        ));
+        );
+    }
+
+    @Test
+    public void test_version_created_is_read_from_partitioned_template() throws Exception {
```
Comment (matriv) [inline server/src/test/java/io/crate/metadata/doc/DocTableInfoTest.java:230]: Do we also have a test to check that `Version.CURRENT` is used without providing customSettings?

**Review by matriv — COMMENTED** [review]
Left a few comments/questions.

Do we need to add a bwc test in `crate-qa` to validate the upgrade?

**Comment by seut** [comment]
@matriv 
> Do we need to add a bwc test in crate-qa to validate the upgrade?

What to test there exactly? Nothing is upgraded, so not sure what we should test additionally...

**Review by matriv — APPROVED** [review]
Thx @seut!

**Comment by matriv** [comment]
> What to test there exactly? Nothing is upgraded, so not sure what we should test additionally...

Sorry, my bad, I somehow thought that the issue took place when a partitioned table created pre v5.0.0 was upgraded to v5.0.0.

**Pull request merged**